scripts/quests/ahtUrhgan/Rat_Race.lua

```diff
@@ -0,0 +1,212 @@
+                        quest:getVar(player, 'Prog') == 4 and
+                        trade:hasItemQty(xi.items.AHTAPOT, 1) and
+                        trade:hasItemQty(xi.items.ISTAKOZ, 1) and
+                        trade:hasItemQty(xi.items.ISTAVRIT, 1) and
+                        trade:hasItemQty(xi.items.ISTIRIDYE, 1) and
+                        trade:hasItemQty(xi.items.MERCANBALIGI, 1) and
+                        trade:getItemCount() == 5
+                    then
+                        return quest:progressEvent(310)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 3 then
+                        return quest:progressEvent(309)
+                    elseif questProgress == 4 then
+                        return quest:event(242)
+                    elseif questProgress >= 5 then
+                        return quest:event(315)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [309] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 4)
+                end,
+
+                [310] = function(player, csid, option, npc)
+                    if npcUtil.giveItem(player, xi.items.BOWL_OF_NASHMAU_STEW) then
+                        player:tradeComplete()
+                        quest:setVar(player, 'Prog', 5)
+                    end
+                end,
+
+                [311] = function(player, csid, option, npc)
+                    player:tradeComplete()
+                    quest:setVar(player, 'Prog', 6)
+                end,
+
+                [312] = function(player, csid, option, npc)
+                    -- TODO: Modify npcUtil.giveItem to support multiple reward/unable to obtain messages.
+                    if player:getFreeSlotsCount() <= 2 then
+                        player:messageSpecial(nashmauID.text.ITEM_CANNOT_BE_OBTAINEDX, xi.items.IMPERIAL_GOLD_PIECE, 2)
+                        player:messageSpecial(nashmauID.text.ITEM_CANNOT_BE_OBTAINEDX, xi.items.IMPERIAL_MYTHRIL_PIECE, 2)
+                        player:messageSpecial(nashmauID.text.ITEM_CANNOT_BE_OBTAINEDX, xi.items.IMPERIAL_SILVER_PIECE, 3)
+                    else
+                        player:addItem(xi.items.IMPERIAL_GOLD_PIECE, 2)
+                        player:addItem(xi.items.IMPERIAL_MYTHRIL_PIECE, 2)
+                        player:addItem(xi.items.IMPERIAL_SILVER_PIECE, 3)
+                        player:messageSpecial(nashmauID.text.ITEM_OBTAINEDX, xi.items.IMPERIAL_GOLD_PIECE, 2)
+                        player:messageSpecial(nashmauID.text.ITEM_OBTAINEDX, xi.items.IMPERIAL_MYTHRIL_PIECE, 2)
+                        player:messageSpecial(nashmauID.text.ITEM_OBTAINEDX, xi.items.IMPERIAL_SILVER_PIECE, 3)
+                        quest:complete(player)
```

Comment:
Complete function can handle multiple items in reward section if used in {}

Reply:
Complete function doesn't handle the messages properly at this time.  It defaults to only `ITEM_OBTAINED` instead of `ITEM_OBTAINEDX` if qty > 1.

Reply:
In that case 
If npcUtil.giveItem {} then
Quest:complete()

Reply:
That's the problem.  npcUtil.giveItem doesn't handle the quantities

Reply:
Will update this as soon as the npcUtil changes make it in